### ISSUE TITLE
repart: maybe add Format=empty which sets partition label to "_empty" and NoAuto=1, and little else, for compat with sysupdate

### ISSUE THREAD

**Issue by NekkoDroid** [systemd/systemd#34596]
### systemd version the issue has been seen with

256

### Used distribution

Arch Linux

### Linux kernel version used

6.10.10-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd-dissect

### Expected behaviour you didn't see

When I `sd-repart` a live "installer image" (built from https://github.com/NekkoDroid/NekkOS/tree/f898a9efea1dfc65d04c3a2c077d25fc13baba89) to a temporary file `image.raw`, it creates 2 `usr` and `usr-verity` partitions (for A/B updates).

`systemd-firstboot --image=image.raw --prompt` should successfully run and set up the new partitions

### Unexpected behaviour you saw

When attempting to run `sd-firstboot --image=image.raw` the command fails with `Failed to mount image: File system type not supported or not known.`, presumably from detecting the `_empty` `usr` partition with no filesystem and attempting to mount it.

Similarly `systemd-dissect` only displays the 2 `_empty` `usr` and `usr-verity` partitions with the actual populated `usr` and `usr-verity` partitions nowhere to be seen.

Logs: https://gist.github.com/NekkoDroid/35426d472aba9e7cbaa01cb088de74fa

### Steps to reproduce the problem

1. Build linked image & run in VM
1. `systemd-repart --dry-run=false --defer-partitions=var --size=20G --empty=create image.raw`
1. `systemd-firstboot --image=image.raw --prompt --setup-machine-id` <- failure
1. `systemd-dissect image.raw` <- missing partitions

(Same happens when `--defer-partitions` is omitted)

**Comment by poettering** [comment]
Use NoAuto= in repart.d to ensure the empty partitions are not considered when dissecting.

**Comment by NekkoDroid** [comment]
That indeed works around the problem I am facing, so my immediate issue is solved. 

But I do think that either repart should set the `NoAuto=` automatically somehow, e.g. a new `Format=empty` that implies `NoAuto=true` and `Label=_empty` (and maybe some other things?). 

In the `sd-firstboot` the dissection logic should be able to handle this kinda case more gracefully and attempt to mount the "newest" partition according to the labeling instead of just attempting to mount the first it finds (if my assumption is even correct).

More complicated would be attempting to read the `roothash` or `usrhash` cmdline of the "newest" loader entry and mounting that corresponding partition.

**Comment by poettering** [comment]
Yeah, I think it would make sense to add a bit of magic to repart there. 

Not sure what it should be though. Could be Format=empty, or it could just be that we imply NoAuto= in case Label=_empty is set and NoAuto is not set otherwise.



> In the `sd-firstboot` the dissection logic should be able to handle this kinda case more gracefully and attempt to mount the "newest" partition according to the labeling instead of just attempting to mount the first it finds (if my assumption is even correct).

It actually does that already: for all partitions of the right type where the noauto flag is off it will do a strverscmp() (well, not really, it's a derivative of that function), so that the nwest OS is used.

**Comment by NekkoDroid** [comment]
> It actually does that already: for all partitions of the right type where the noauto flag is off it will do a strverscmp() (well, not really, it's a derivative of that function), so that the nwest OS is used.

Weird that it then attempts to mount the `_empty` partitions. When I size the image so that the empty partitions are left out I don't encounter that problem.

**Comment by poettering** [comment]
"_" is after the letters in the ASCII table, hence _empty orders after proper GPT partition label strings, and this is considered newer by strverscmp(), hence is preferred. Everyhing hence really works as it should.
